### PR TITLE
Added shell sort in C

### PR DESCRIPTION
**Pull request by Mithunitp**
For issue #57 .

### CODE REVIEW

**Comment by ankitapuri** [comment]
> For issue #57 .

Explain your code with comments

File: Shell_Sort/shell_sort.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        // Do a gapped insertion sort for this gap size(i). 
+        // The first gap elements a[0..i-1] are already in gapped order 
+        // keep adding one more element until the entire array is gap sorted  
+        for (j = i; j < size; j++)
+        {
+            // add a[j] to the elements that have been gap sorted 
+            // save a[j] in temp and make space at position j
+            
+            temp = arrary[j];
```
Comment (ankitapuri) [inline Shell_Sort/shell_sort.c]: Correct the declaration

File: Shell_Sort/shell_sort.c [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    array[k]=temp;
+
+            }
+        }
+    }
+}
+
+int main()
+{
+    int array[30];
```
Comment (ankitapuri) [inline Shell_Sort/shell_sort.c]: Make the array dynamic 
Its size should depend on user input

Comment (Mithunitp) [reply Shell_Sort/shell_sort.c]: Ok will do

File: Shell_Sort/shell_sort.c (hunk 1 above)
Comment (Mithunitp) [reply Shell_Sort/shell_sort.c]: Corrected

File: Shell_Sort/shell_sort.c (hunk 2 above)
Comment (Mithunitp) [reply Shell_Sort/shell_sort.c]: Declared the array with user defined size.

**Review by ankitapuri — APPROVED** [review]

**Pull request merged**